src/wp-includes/block-supports/elements.php

```diff
@@ -166,8 +166,8 @@ function wp_render_elements_support_styles( $pre_render, $block ) {
 			'skip'     => $skip_button_color_serialization,
 		),
 		'link'    => array(
-			'selector'       => ".$class_name a",
-			'hover_selector' => ".$class_name a:hover",
+			'selector'       => ".$class_name a:where(:not(.wp-element-button))",
+			'hover_selector' => ".$class_name a:where(:not(.wp-element-button)):hover",
```

Comment:
Why now use `a:not(.wp-element-button)`?

Reply:
Thanks for the question @mukeshpanchal27, I take it you mean "why **not** use `:not`"?

This selector uses the `:where` to reduce specificity and make it easier for themes to override the link element styles as desired.

In the case of this PR, it's a bug fix that prevents the misapplication of the link element styles to buttons but by using the same selector as theme.json it also avoids increasing the specificity. This keeps the ease of overriding styles if a theme desires while still resulting in the expected behaviour.

The original PR introducing the lowered specificity for link element styles in theme.json can be found in https://github.com/WordPress/gutenberg/pull/42669. That was backported in https://github.com/WordPress/wordpress-develop/pull/3206.